DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -1971,15 +1972,25 @@ internal async Task<DiscordGuild> GetGuildAsync(ulong guildId, bool? with_counts
             }
         }
 
-        internal async Task<DiscordRole> ModifyGuildRoleAsync(ulong guild_id, ulong role_id, string name, Permissions? permissions, int? color, bool? hoist, bool? mentionable, string reason)
+        internal async Task<DiscordRole> ModifyGuildRoleAsync(ulong guild_id, ulong role_id, string name, Permissions? permissions, int? color, bool? hoist, bool? mentionable, string reason, Stream icon, string emoji)
         {
+            string image = null;
+
+            if (icon != null)
+            {
+                using var it = new ImageTool(icon);
+                image = it.GetBase64();
+            }
+
             var pld = new RestGuildRolePayload
             {
                 Name = name,
                 Permissions = permissions & PermissionMethods.FULL_PERMS,
                 Color = color,
                 Hoist = hoist,
-                Mentionable = mentionable
+                Mentionable = mentionable,
+                Emoji = emoji,
```

Comment:
I'm not sure whether Discord supports default emojis for role icons, but it might be nice to add a check whether it's an image emote or a default emote. Also, I don't think the user should be able to set both icon and image... maybe a check for that too?
(with exceptions of course. InvalidArgumentException should be fine I think)

Reply:
Image emotes get set as the icon, where as unicode emotes use `emoji`.

Reply:
@Naamloos I hope [this comment](https://github.com/DSharpPlus/DSharpPlus/pull/1099#discussion_r717215330)

Reply:
Was resolved on Discord.